### PR TITLE
add custom web fonts

### PR DESCRIPTION
**Pull request by sveroude**
- add custom web fonts;
- load web fonts asynchronously using `fontfaceobserver`;
- use `sessionstorage` to set flag after fonts load for the first time

### CODE REVIEW

File: src/templates/product.html [hunk 1]
```diff
@@ -66,5 +66,52 @@ <h1 class="page-header-title">{{ title | safe }}</h1>
...
+
+            if (sessionStorage.fontsLoaded) {
+                html.classList.add(FONTS_LOADED_CLASS);
+            } else {
+                var script = document.createElement('script');
+                script.src = 'https://rawgit.com/voorhoede/voorhoede-ocelot-formatter/fonts/src/assets/scripts/fontfaceobserver.standalone.js';
+                script.async = true;
+
+                script.onload = function () {
+                    var ralewayMedium = new window.FontFaceObserver('Raleway', {weight: 400});
```
Comment (jbmoelker) [inline src/templates/product.html]: Verbose, rather split config from code?
first:

```js
var FONTS_LOADED_CLASS = 'fonts-loaded';
var fonts = {
  Noto: [ { weight: 400 }, { weight: 400, style: 'italic' }, { weight: 700 }],
  Raleway: [ { weight: 400 }, { weight: 600 }, { weight: 700 }]
}
```
and later
```js
function loadFonts(fonts, fontsLoadedClass) {
  // iterate over `fonts`, etc
}
```

File: src/assets/index.css [hunk 2]
```diff
@@ -1,3 +1,58 @@
+@font-face {
+    font-family: 'Raleway';
+    src: url('https://rawgit.com/voorhoede/voorhoede-ocelot-formatter/fonts/src/assets/fonts/raleway-extrabold-webfont.woff2') format("woff2"),
```
Comment (jbmoelker) [inline src/assets/index.css]: I would use local font notation in source code and use a build task to replace with remote url during deploy.

So in source, something like:
```css
@font-face {
  font-family: 'Raleway';
  src: url('assets/fonts/raleway-extrabold-webfont.woff2') format("woff2"),
         url('assets/fonts/raleway-extrabold-webfont.woff') format("woff");
  font-weight: 700;
  font-style: normal;
}
```
And in build replace `assets/fonts/` with `https://rawgit.com/voorhoede/voorhoede-ocelot-formatter/fonts/src/assets/fonts/` or `https://voorhoede.github.io/voorhoede-ocelot-formatter/assets/fonts/`.

File: src/templates/product.html [hunk 3]
```diff
@@ -66,5 +66,52 @@ <h1 class="page-header-title">{{ title | safe }}</h1>
...
+                document.head.appendChild(script);
+            }
+        }());
+    </script>
+    <script>
+        (function(window) {
+            var html = document.documentElement;
+            var FONTS_LOADED_CLASS = 'fonts-loaded';
+
+            if (sessionStorage.fontsLoaded) {
```
Comment (jbmoelker) [inline src/templates/product.html]: move the `fontsLoaded` check to the `<head>` to prevent the FOUT on repeat views.

**Comment by jbmoelker** [comment]
@sveroude the font `src` urls are still an issue as they point to the files on the `fonts` branch via rawgit. But this `fonts` branch will be removed once this PR is merged, which means the font files will then no longer be available. So please still make the changes as suggested in this review: https://github.com/voorhoede/voorhoede-ocelot-formatter/pull/39/files#r94327452

File: src/assets/index.css (hunk 2 above)
Comment (sveroude) [inline src/assets/index.css]: ✅

**Pull request merged**